bubble_sort.rb

```diff
@@ -0,0 +1,25 @@
+def bubble_sort(arr)
+  arr.each_index do |_|
+    arr.each_index do |i|
+      arr[i], arr[i + 1] = arr[i + 1], arr[i] if i < arr.length - 1 && arr[i] > arr[i + 1]
+    end
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  arr.each_index do |_|
+    arr.each_index do |i|
+      if i < arr.length - 1
+        arr[i], arr[i + 1] = arr[i + 1], arr[i] if yield(arr[i], arr[i + 1]).positive?
+      end
+    end
+  end
+  arr
+end
+
+result = bubble_sort_by(['hi', 'hello', 'hey', 'hand', 'house', 'houze with z']) do |left, right|
+  left.length - right.length
+end
+
+p result
```

Comment:
Let's delete these lines. They are considered as debugging partials. I see that you are testing your code here which shows you are trying to be sure what you are doing. But we don't need it here. :pray: